tests/integration/collection_description/updates/add/embeddings_test.go

```diff
@@ -0,0 +1,136 @@
+	"testing"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/client"
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestColDescrUpdate_AddVectorEmbeddingWithUnknownFieldName_ShouldError(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						name: String
+						about: String
+						name_v: [Float32!]
+					}
+				`,
+			},
+			testUtils.PatchCollection{
+				Patch: `
+					[
+						{ "op": "add", "path": "/1/Embeddings/-", "value": {"FieldName": "foo"} }
+					]
+				`,
+				ExpectedError: "the given field does not exist. Vector field: foo",
+			},
+		},
+	}
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestColDescrUpdate_AddVectorEmbeddingWithUnknownEmbeddingGenerationField_ShouldError(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						name: String
+						about: String
+						name_v: [Float32!]
+					}
+				`,
+			},
+			testUtils.PatchCollection{
+				Patch: `
+					[
+						{ "op": "add", "path": "/1/Embeddings/-", "value": {"FieldName": "name_v", "Fields": ["name", "foo"]} }
+					]
+				`,
+				ExpectedError: "the given field does not exist. Embedding generation field: foo",
+			},
+		},
+	}
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestColDescrUpdate_AddVectorEmbeddingWithInvalidEmbeddingGenerationFieldKind_ShouldError(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						name: String
+						custom: JSON
+						name_v: [Float32!]
+					}
+				`,
+			},
+			testUtils.PatchCollection{
+				Patch: `
+					[
+						{ "op": "add", "path": "/1/Embeddings/-", "value": {"FieldName": "name_v", "Fields": ["name", "custom"]} }
+					]
+				`,
+				ExpectedError: "invalid field type for vector embedding generation. Actual: JSON",
+			},
+		},
+	}
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestColDescrUpdate_AddVectorEmbedding_ShouldSucceed(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						name: String
+						about: String
+						name_v: [Float32!]
+					}
+				`,
+			},
+			testUtils.PatchCollection{
+				Patch: `
+					[
+						{ "op": "add", "path": "/1/Embeddings/-", "value": {"FieldName": "name_v", "Fields": ["name", "about"], "Provider": "ollama", "Model": "nomic-embed-text",  "URL": "http://localhost:11434/api"} }
+					]
+				`,
+			},
+			testUtils.GetCollections{
+				ExpectedResults: []client.CollectionDescription{
+					{
+						Name:           immutable.Some("Users"),
+						IsMaterialized: true,
+						Embeddings: []client.EmbeddingDescription{
+							{
+								FieldName: "name_v",
+								Fields:    []string{"name", "about"},
+								Provider:  "ollama",
+								Model:     "nomic-embed-text",
+								URL:       "http://localhost:11434/api",
```

Comment:
todo: Please add tests covering the omission of each property in a patch, and SDL ( e.g. what happens if the user does not provide a `url`?).